test/e2e/workflow/deployment_test.go

```diff
@@ -89,6 +89,9 @@ var _ = Describe("NetworkAddonsConfig", func() {
 				[]Component{MacvtapComponent},
 			),
 		)
+		It("should deploy prometheus if NetworkAddonsConfigSpec is not empty", func() {
+			testConfigCreate(gvk, cnao.NetworkAddonsConfigSpec{MacvtapCni: &cnao.MacvtapCni{}}, []Component{MacvtapComponent, MonitoringComponent})
```

Comment:
why actually not to deploy prometheus even if spec is empty ?
some of the gauges might still need to be available aren't they ?

Reply:
it's simply the way CNAO deploys components, it only does it if the CR is there and not empty.